org.civicrm.reqangular/src/common/angular-date/src/services/DateFactory.js

```diff
@@ -20,13 +23,33 @@ function DateFactory() {
                 ];
             }
 
-            if(typeof strict === 'undefined'){
+            if (typeof strict === 'undefined') {
                 strict = true;
             }
 
             return moment(dateString, format, strict);
+        },
+
+        /**
+         * @description Returns Current Date Format
+         * @returns {string}
+         */
+        getDateFormat: function () {
+            return this.dateFormat;
+        },
+
+        /**
+         * @description Fetches date format setting from
+         * @returns {Promise}
+         */
+        fetchDateFormatFromSettings: function(){
```

Comment:
Actually this method should be synchronous. That's why I decided to delegate async call to another method. Also, I'd need to return string or a promise object depending on this condition - that wouldn't be good idea. 
It's better that function return one exact type.

Also these functions do completely different things - one fetches data from server, another delivers it to the user.